datahub-web-react/src/app/ingest/source/builder/RecipeForm/ConnectionRequest.tsx

```diff
@@ -0,0 +1,266 @@
+    DOMAINS = 'Domains',
+    DATA_PROFILING = 'Data Profiling',
+    USAGE_STATS = 'Dataset Usage',
+    PARTITION_SUPPORT = 'Partition Support',
+    DESCRIPTIONS = 'Descriptions',
+    LINEAGE_COARSE = 'Table-Level Lineage',
+    LINEAGE_FINE = 'Column-level Lineage',
+    OWNERSHIP = 'Extract Ownership',
+    DELETION_DETECTION = 'Detect Deleted Entities',
+    TAGS = 'Extract Tags',
+    SCHEMA_METADATA = 'Schema Metadata',
+    CONTAINERS = 'Asset Containers',
+}
+
+interface ConnectionCapability {
+    capable: boolean;
+    failure_reason: string | null;
+    mitigation_message: string | null;
+}
+
+interface CapabilityReport {
+    [key: string]: ConnectionCapability;
+}
+
+interface TestConnectionResult {
+    internal_failure?: boolean;
+    internal_failure_reason?: string;
+    basic_connectivity?: ConnectionCapability;
+    capability_report?: CapabilityReport;
+}
+
+const BASIC_CONNECTIVITY_SUCCESS = "You're able to connect to this source with your given recipe.";
+
+export function getRecipeJson(recipeYaml: string) {
+    // Convert the recipe into it's json representation, and catch + report exceptions while we do it.
+    let recipeJson;
+    try {
+        recipeJson = yamlToJson(recipeYaml);
+    } catch (e) {
+        const messageText = (e as any).parsedLine
+            ? `Please fix line ${(e as any).parsedLine} in your recipe.`
+            : 'Please check your recipe configuration.';
+        message.warn(`Found invalid YAML. ${messageText}`);
+        return null;
+    }
+    return recipeJson;
+}
+
+interface Props {
+    recipe: string;
+}
+
+function ConnectionRequest(props: Props) {
+    const { recipe } = props;
+    const [isLoading, setIsLoading] = useState(false);
+    const [isModalVisible, setIsModalVisible] = useState(false);
+    const [pollingInterval, setPollingInterval] = useState<null | NodeJS.Timeout>(null);
+    const [testConnectionResult, setTestConnectionResult] = useState<null | TestConnectionResult>(null);
+    const [createTestConnectionRequest, { data: requestData }] = useCreateTestConnectionRequestMutation();
+    const [getIngestionExecutionRequest, { data: resultData, loading }] = useGetIngestionExecutionRequestLazyQuery();
+
+    useEffect(() => {
+        if (requestData && requestData.createTestConnectionRequest) {
+            const interval = setInterval(
+                () =>
+                    getIngestionExecutionRequest({
+                        variables: { urn: requestData.createTestConnectionRequest as string },
+                    }),
+                2000,
+            );
+            setIsLoading(true);
+            setIsModalVisible(true);
+            setPollingInterval(interval);
+        }
+    }, [requestData, getIngestionExecutionRequest]);
+
+    useEffect(() => {
+        if (!loading && resultData) {
+            const result = resultData.executionRequest?.result;
+            if (result && result.status !== RUNNING) {
```

Comment:
Are there any other statuses we can check?

Reply:
yup it could be SUCCESS, FAILURE, or CANCELLED (not sure how we get into a CANCELLED state). I should probably add some state for the FAILURE case at the least! tbh I'm not positive what gets us into that state and if we still get a `structuredReport` - I doubt it but can confirm. But if it fails I can just display a generic "Something went wrong" error message or something.